lib/new_relic/manticore/instrumentation.rb

```diff
@@ -21,12 +21,9 @@ def self.create_segment?
       state = NewRelic::Agent::TransactionState.tl_get
       return false unless state && state.current_transaction
 
-      existing_segments = state.current_transaction.segments
-
-      existing_segments.empty? ||
-        !existing_segments.last.is_a?(
-          ::NewRelic::Agent::Transaction::DatastoreSegment
-        )
+      !state.current_transaction.current_segment.is_a?(
```

Comment:
It looked like there's always a root segment, but we might still want to verify the presence of `current_segment` beforehand?!

Reply:
should not be necessary since there is a root segment and if there is no transaction at all we return earlier

Reply:
added it to be safe